src/Features/LanguageServer/CommonLanguageServerProtocol.Framework/IRequestHandler.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+#nullable enable
+
+namespace CommonLanguageServerProtocol.Framework;
+
+/// <summary>
+/// Top level type for LSP request handler.
+/// </summary>
+public interface IRequestHandler
+{
+    /// <summary>
+    /// Whether or not the solution state on the server is modified
+    /// as a part of handling this request.
+    /// </summary>
+    bool MutatesSolutionState { get; }
+
+    /// <summary>
+    /// Whether or not the handler execution queue should build a solution that represents the LSP
```

Comment:
As mentioned in the meeting, this can't possibly happen without document management being part of CLaSP, so I would guess this shouldn't be here.